[update-webkit] Use auto stash by default when updating repository

zakariaridouh opened the pull request:
#### 17f0dd7e5143cd2782eaf2c3cd553feb4d663f1a
<pre>
[update-webkit] Use auto stash by default when updating repository
<a href="https://bugs.webkit.org/show_bug.cgi?id=300157">https://bugs.webkit.org/show_bug.cgi?id=300157</a>
<a href="https://rdar.apple.com/161933957">rdar://161933957</a>

Reviewed by Alexey Proskuryakov.

We should pass `--autostash` to git pull when updating webkit through the `update-webkit` script.

* Tools/Scripts/webkitdirs.pm:
(determineSourceDir):
(determineXcodeDestination):
(determineXcodeSDKPlatformName):
(XcodeOptions):
(checkBuild):
(findMatchingArguments):
(buildXcodeScheme):
(vcpkgArgsFromFeatures):
(relaunchIOSSimulator):
(iosSimulatorDeviceByUDID):
(debugMiniBrowser):
(runGitUpdate):

Canonical link: <a href="https://commits.webkit.org/301034@main">https://commits.webkit.org/301034@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6b4d6314e9a30caabf0756cf198a0de2a04f14df

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124573 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44246 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34979 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131408 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76525 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/dfd9b962-d2eb-473f-be1e-5b2b08eced24) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126450 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44952 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52819 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94765 "Failed to checkout and rebase branch from PR 51797") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62842 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e07db2f1-22e2-47dc-a161-ee3a0f6238a7) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127527 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35837 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111404 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75338 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/12941753-ecc2-45c1-a812-b79dafc30185) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34774 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29558 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74891 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/116681 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105592 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29790 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134077 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123094 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51428 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39248 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103244 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51836 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107626 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103023 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26262 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48383 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26651 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48362 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51299 "Built successfully") | | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/156260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50702 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/156260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54058 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52388 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Quoting myself from 2023:

> Is the problem that update-webkit doesn't do --autostash? It probably should, I'd r+ such a PR quicker than git purists could object :)

zakariaridouh committed:
[update-webkit] Use auto stash by default when updating repository
https://bugs.webkit.org/show_bug.cgi?id=300157
rdar://161933957

Reviewed by Alexey Proskuryakov.

We should pass `--autostash` to git pull when updating webkit through the `update-webkit` script.

* Tools/Scripts/webkitdirs.pm:
(determineSourceDir):
(determineXcodeDestination):
(determineXcodeSDKPlatformName):
(XcodeOptions):
(checkBuild):
(findMatchingArguments):
(buildXcodeScheme):
(vcpkgArgsFromFeatures):
(relaunchIOSSimulator):
(iosSimulatorDeviceByUDID):
(debugMiniBrowser):
(runGitUpdate):

Canonical link: https://commits.webkit.org/301034@main
